qiskit_ibm/ibm_provider.py

```diff
@@ -902,19 +789,21 @@ def services(self) -> Dict:
         """Return all available services.
 
         Returns:
-            All services available to this provider.
+            All services available to this account.
         """
         return {key: val for key, val in self._services.items() if val is not None}
 
-    def __eq__(
+    def _get_hgp_by_service_and_backend_name(
             self,
-            other: Any
-    ) -> bool:
-        if not isinstance(other, IBMProvider):
-            return False
-        return self.credentials == other.credentials
+            service_name: str,
+            backend_name: str
+    ) -> HubGroupProject:
+        hgps = self._get_hgps()
+        for hgp in hgps:
+            if hgp.has_service(service_name) and \
+                    hgp.get_backend(backend_name):
+                return hgp
+        raise IBMNotAuthorizedError(f"You are not authorized to use {service_name} service.")
```

Comment:
This is not exactly true, e.g. I may have access to the `runtime` service but not the `ibmq_foo` backend.